c++/include/orc/Vector.hh

```diff
@@ -53,6 +53,10 @@ namespace orc {
     // whether the vector batch is encoded
     bool isEncoded;
 
+    // whether the vector is not a numeric vector
+    // or it's created with the option useTightNumericVector
+    bool isTight = true;
```

Comment:
Is it possible not to add this? The reader can get this information by the provided `ReaderOption`

Reply:
But, what if the ColumnVectorBatch is created by another reader which was createed without the useTightNumericVector option?

Reply:
Does the current `ColumnReader` have same issue?

Reply:
> Does the current `ColumnReader` have same issue?

Yes, I think there is.

Reply:
What probably need to fix that as well.

What about adding a new function to `ColumnVectorBatch` like `int32_t fixed_type_size() const` to return the width in bytes for fixed length type? For variable length type or nested type we can return -1.

Reply:
> What probably need to fix that as well.
> 
> What about adding a new function to `ColumnVectorBatch` like `int32_t fixed_type_size() const` to return the width in bytes for fixed length type? For variable length type or nested type we can return -1.

I think what we are going to do is handle incorrect usage by users (such as passing a vector that was created by other readers). However, we don’t know what causes the failure of casting. The vector may have been created by another reader with a different read type from the current reader, or it may have been created by another reader with a different `useTightNumericVector` option. `fixed_type_size()` may not help us infer what the user did wrong, but with `isTight`, we could suggest that the user check their options.

Reply:
OK, that sounds good. 

However, `isTight` here is a little bit confusing since it applies to only numeric types. We need to be careful to add any field to the public api. And it should also be used with its type together. 

In the `ColumnReader`, we always check the result of dynamic_cast of the vector batch. Is it enough if we can use dynamic_cast and check if it is not null in the `ConvertColumnReader`?

Reply:
> OK, that sounds good.
> 
> However, `isTight` here is a little bit confusing since it applies to only numeric types. We need to be careful to add any field to the public api. And it should also be used with its type together.
> 
> In the `ColumnReader`, we always check the result of dynamic_cast of the vector batch. Is it enough if we can use dynamic_cast and check if it is not null in the `ConvertColumnReader`?

I think that`s OK, so we will only verify the `ReaderOption` and check the result of dynamic_cast in `ConvertColumnReader`.